state-chain/chains/src/sol/sol_tx_core.rs

```diff
@@ -1227,49 +1318,62 @@ mod tests {
 
 	#[test]
 	fn create_ccm_native_transfer() {
-		let durable_nonce = Hash::from_str("FJzAoeurcnAKG7eNFhzixySntXkDzoEh2bcRNfKm1gsy").unwrap();
+		let durable_nonce = Hash::from_str(TEST_DURABLE_NONCE).unwrap();
 		let agg_key_keypair = Keypair::from_bytes(&RAW_KEYPAIR).unwrap();
 		let agg_key_pubkey = agg_key_keypair.pubkey();
-		let to_pubkey = Pubkey::from_str("pyq7ySiH5RvKteu2vdXKC7SNyNDp9vNDkGXdHxSpPtu").unwrap();
-		let cf_receiver = Pubkey::from_str("NJusJ7itnSsh4jSi43i9MMKB9sF4VbNvdSwUA45gPE6").unwrap();
-		let amount: u64 = TRANSFER_AMOUNT;
+		let to_pubkey = Pubkey::from_str(TRANSFER_TO_ACCOUNT).unwrap();
+		let extra_accounts = ccm_extra_accounts();
+		let ccm_parameter = ccm_parameter();
 
 		let instructions = [
 			SystemProgramInstruction::advance_nonce_account(
 				&Pubkey::from_str(NONCE_ACCOUNTS[0]).unwrap(),
 				&agg_key_pubkey,
 			),
-			SystemProgramInstruction::transfer(&agg_key_pubkey, &to_pubkey, amount),
+			ComputeBudgetInstruction::set_compute_unit_price(COMPUTE_UNIT_PRICE),
+			ComputeBudgetInstruction::set_compute_unit_limit(COMPUTE_UNIT_LIMIT),
+			SystemProgramInstruction::transfer(&agg_key_pubkey, &to_pubkey, TRANSFER_AMOUNT),
 			ProgramInstruction::get_instruction(
 				&VaultProgram::ExecuteCcmNativeCall {
-					source_chain: 1,
-					source_address: vec![11u8, 6u8, 152u8, 22u8, 3u8, 1u8],
-					message: vec![124u8, 29u8, 15u8, 7u8],
-					amount,
+					source_chain: ccm_parameter.source_chain as u32,
+					source_address: codec::Encode::encode(&ccm_parameter.source_address),
+					message: ccm_parameter.channel_metadata.message.to_vec(),
+					amount: TRANSFER_AMOUNT,
 				},
 				Pubkey::from_str(VAULT_PROGRAM).unwrap(),
 				vec![
-					AccountMeta::new_readonly(
-						Pubkey::from_str(VAULT_PROGRAM_DATA_ACCOUNT).unwrap(),
-						false,
-					),
-					AccountMeta::new_readonly(agg_key_pubkey, true),
-					AccountMeta::new(to_pubkey, false),
-					AccountMeta::new_readonly(cf_receiver, false),
-					AccountMeta::new_readonly(Pubkey::from_str(SYSTEM_PROGRAM_ID).unwrap(), false),
-					AccountMeta::new_readonly(
-						Pubkey::from_str(SYS_VAR_INSTRUCTIONS).unwrap(),
-						false,
-					),
-				],
+					vec![
```

Comment:
This looks incorrect to me but I could be wrong.
This should be a vector of AccountMetas where the extra accounts are appended at the end:
vec![a,b...,remainingAccount0, remainingAccount1...]
However, what I'm reading here is this:
vec![vec![a,b...],vec![remainingAccount0, remainingAccount1...]]

Reply:
the flatten call below flattens the 2 vec into a single one. But I understands that this is not obvious. For the tests in the Core, we should hardcode the value/structure to ensure the instruction builder builds the correct tx hash. 
I'll change it to a more direct value.

Reply:
if the flatten does exactly that then it's fine as is.